Core/Interfaces/IProductService.cs

```diff
@@ -0,0 +1,17 @@
+﻿using LearnNet_CatalogService.Core.DTO;
+
+namespace LearnNet_CatalogService.Core.Interfaces
+{
+    public interface IProductService
+    {
+        Task<IList<ProductDTO>> GetAllProductsAsync();
+
+        Task<ProductDTO?> GetProductByIdAsync(int id);
```

Comment:
why is nullable annotation required here?

Reply:
Because product could be not found